js/components/generate/GenerateForm.js

```diff
@@ -280,84 +298,92 @@ export default class GenerateForm extends React.Component {
               includeRaws: false,
               includeGrades: false,
             }}
+            validate={this.validateForm}
+            validateOnChange={false}
+            validateOnBlur={false}
             onSubmit={this.handleSubmit}
           >
-            <Form id="generateForm">
-              <Field
-                id="courses"
-                name="courses"
-                type="text"
-                placeholder="e.g., CSC207H1, CSC324H1"
-              />
-
-              <h2 id="filter-title">Optional filters</h2>
-
-              <label htmlFor="departments">
-                Only include courses these departments
-              </label>
-              <Field
-                id="departments"
-                name="departments"
-                type="text"
-                placeholder="Enter 3-letter department codes separated by commas"
-                style={{ marginBottom: "1em" }}
-              />
-
-              <label htmlFor="taken">Do not show these courses</label>
-              <Field
-                id="taken"
-                name="taken"
-                type="text"
-                style={{ marginBottom: "1em" }}
-                placeholder="E.g., CSC207H1, CSC236H1"
-              />
-
-              <label htmlFor="maxDepth">
-                Depth of prerequisite chain (0 shows all prerequisites)
-              </label>
-              <p>
+            {({ errors }) => (
+              <Form id="generateForm">
                 <Field
-                  id="maxDepth"
-                  name="maxDepth"
-                  type="number"
-                  min="0"
-                  step="1"
-                  style={{ marginBottom: "1em" }}
+                  id="courses"
+                  name="courses"
+                  type="text"
+                  placeholder="e.g., CSC207H1, CSC324H1"
                 />
-              </p>
-
-              {/* <label htmlFor="location">Campus</label>
-              <Field id="location" name="location" as="select" multiple
-                style={{ verticalAlign: 'text-top', marginLeft: '1em', marginBottom: '1em', color: 'black' }}>
-                <option value="utsg">St. George</option>
-                <option value="utm">Mississauga</option>
-                <option value="utsc">Scarborough</option>
-              </Field>
-
-              <p>
-                <label htmlFor="includeRaws">Include non-course prerequisites</label>
-                <Field id="includeRaws" name="includeRaws" type="checkbox"
-                  style={{ marginLeft: '1em', verticalAlign: 'middle' }}
+                <ErrorMessage id="error-message" name="courses" component="div" />
```

Comment:
In general, HTML `id`s should not be duplicated. You should instead use classes to apply common styling. In JSX syntax you need to use `className` to set the class.